bot/cogs/grades_cog.py

```diff
@@ -0,0 +1,259 @@
+            C.append(i['C'])
+            D.append(i['D'])
+            F.append(i['F'])
+            W.append(i['W'])
+        gradesList = (A,B,C,D,F,W,P,NP)
+        for letterGrade in gradesList:
+            for i in range(len(letterGrade)):
+                letterGrade[i] = int(letterGrade[i][:-1])
+
+        AvgA = sum(A)//len(A)
+        AvgB = sum(B)//len(B)
+        AvgC = sum(C)//len(C)
+        AvgD = sum(D)//len(D)
+        AvgF = sum(F)//len(F)
+        AvgWithdraw = sum(W)//len(W)
+
+        return  [(AvgA + AvgB), (AvgF+AvgWithdraw), (len(data_list))]
+
+    def toCamelCase(self, Name):
+        items = Name.split()
+        for i in range(len(items)):
+            items[i] = items[i][0].upper() + items[i][1:].lower()
+        return " ".join(items)
+
+    def getAllCourses(self):
+        temp = []
+        for i in self.master_list:
+            if i[:-5] not in temp:
+                temp.append(i[:-5])
+            else:
+                pass
+        with open("bot/cogs/Student-Teacher/assets/CourseList.txt", "w") as f:
+            for i in range(len(temp)):
+                f.write(temp[i]+"\n")
+
+    def searchCourse(self, query):
+        # query = input("What course do you want to search? (Format: <Course>-<number> ie CPSC-3720)\n")
+        string = ""
+        # try:
+            
+        # string += "\n--------------------------------------------------------\n"
+        string += "Since Spring 2014, there was an average of " + self.process_Search(query)
+        string += "DISCLAIMER:\n"
+        string += "Not every professor listed will be at Clemson, this is a tool built for better information but not complete information\n"
+        # string += "Take it at your own discression\n"
+        # string += "\nIn addition, this system works on the Grade Distribution Releases located at https://www.clemson.edu/institutional-effectiveness/oir/data-reports/\n"
+        # string += "As a result, the limitations according to the GDR are as follows:\n"
+        string += '*Course Sections that meet the following conditions are not included: Undergraduate classes with less than 10 students or Graduate classes with less than 5 students. In addition, if a section has all but 1 student making a common grade (example: All but one student makes a "B" in a class), the section is excluded.*'
+        # string += '\n----------------------------------------------------------------------------'
+        return string
+        # except NotADirectoryError as e:
+        #     print("Class not found, are you sure you used the correct format? (Ex: cpsc-2120)")
+        
+        # except Exception as e:
+        #     print("Something went wrong somewhere, idk bout that")
+        #     print(e)
+
+    def go(self, course):
+        self.initialize()
+        self.getAllCourses()
+        return self.searchCourse(course)
+
+    @commands.command()
+    async def grades(self, ctx, course):
+        print(course)
```

Comment:
The validation for the input should be done here, before the search is done

Reply:
the validation can't complete because it needs to initialize first.